src/services/__tests__/MatchService.unit.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { BallChaserQueueBuilder } from "../../../.jest/Builder";
+import { createRandomMatch } from "../MatchService";
+import { mocked } from "ts-jest/utils";
+import ActiveMatchRepository from "../../repositories/ActiveMatchRepository";
+import { Team } from "../../types/common";
+import { NewActiveMatchInput } from "../../repositories/ActiveMatchRepository/types";
+
+jest.mock("../../repositories/ActiveMatchRepository");
+jest.mock("../../utils/getEnvVariable");
+
+describe("Match Service tests", () => {
+  const mockBallChasers = BallChaserQueueBuilder.many(6);
+  mockBallChasers.forEach((player) => (player.team = null));
+  const addMatchMock = mocked(ActiveMatchRepository.addActiveMatch);
+
+  it("random teams are created", async () => {
+    await createRandomMatch(mockBallChasers);
+    expect(mockBallChasers.filter((player) => player.team == Team.Orange)).toHaveLength(3);
+  });
+
+  it("updates queue with update player teams", async () => {
+    const activeMatch: Array<NewActiveMatchInput> = [];
+    mockBallChasers.forEach((player) => {
+      if (player.team != null) {
+        activeMatch.push({ id: player.id, team: player.team });
+      }
+    });
+    expect(addMatchMock).toHaveBeenCalled();
```

Comment:
Is this test relying on the one before it to have run?

Reply:
I fixed the tests, I am not sure why you asked about it needing to have run - but since I changed it - can you check.

Reply:
The way you had it before was correct, it just seemed like you were expecting `addMatchMock` in your "updates queue" test even though you weren't calling it in the scope of your test. Which lead me to believe that the "random teams are created" test _had_ to run before the "updates queue" test for it to work properly.

I'm not a huge fan of calling the `createRandomMatch` in the `beforeAll`. Each `it` test should do all the necessary setup, call the function to be tested, and make the expect calls. This makes it a lot easier to figure out what's going on in each test and allows them to run independently of one another.

Reply:
Fixed in next commit.